Metadata: add datasets as CSV

Gero1999 opened the pull request:
## Issue

Closes #760 

## Description
Added using still `usethis` procedures. The CSV file will now be available to compare changes through GitHub and also for users to review. Probably interesting to link for #758 or a new PR in this direction

## Definition of Done
- [x] metadata .csv files in github
- [x] use these files to update metadata .rds files (using `usethis`)

## How to test
From now we only need to change the datasets in data-raw
- Make a change in `data-raw/metadata_nca_X.csv` without commiting
- Run `data-raw/metadata_nca_X.R`
- Run the App and make sure the change is reflected there

## Contributor checklist
- [x] Code passes lintr checks
- [x] Code passes all unit tests
- [x] New logic covered by unit tests
- [x] New logic is documented
- ~Package version is incremented~

## Notes to reviewer

Anything that the reviewer should know before tacking the pull request?

Gero1999 committed:
add data-raw objects with NCA parameters and variables

Gero1999 committed:
add data-raw in .Rbuildignore

Gero1999 committed:
add code to source metadata automatically with every App run

Gero1999 committed:
fix: typpos

Gero1999 committed:
rm code to generate automatically metadata. should be manual every time

Gero1999 committed:
refactor: lintr

js3110 reviewed (approved):
LGTM!

Just a note: need to make sure that if this is merged before #778 then it needs to be updated there, or vice versa!

js3110 commented:
@Gero1999 this just needs to be updated so ADPC -> ADNCA in the nca_variables files

Gero1999 committed:
update metadata_nca_variables (ADPC > ADNCA) to match changes in #778

Gero1999 committed:
merge: conflict metadata_nca_variables.rda without need to solve

Merge remote-tracking branch 'origin/main' into 760-enhancement-add-metadata-not-as-rds

# Conflicts:
#	data/metadata_nca_variables.rda

js3110 commented on data/metadata_nca_variables.rda:
@Shaakon35 thats one of the csvs thats added here
